proj_eclipse/src/main/java/smartmag/models/BoxModel.java

```diff
@@ -83,14 +86,25 @@ protected BoxRecord getRecord() {
 		return record;
 	}
 
+	public BoxModel CreateBox(Box b)
+			throws SQLIntegrityConstraintViolationException {
+		BoxModel bm = getBoxModel(b);
+		if (bm == null)
+			bm = new BoxModel(b);
+		if (bm.isSavedInDb())
+			throw new IllegalArgumentException("box già presente");
+		bm.create();
+		return bm;
+	}
+
 	/**
 	 * permette di creare il record del box nel database, solo se non é giá
 	 * presente
 	 * 
 	 * @throws SQLIntegrityConstraintViolationException se il record é giá
 	 *                                                  presente nel db
 	 */
-	public void createBox() throws SQLIntegrityConstraintViolationException {
+	public void create() throws SQLIntegrityConstraintViolationException {
```

Comment:
privato poiché esiste già createBox publico per creare un Box
```suggestion
	private void create() throws SQLIntegrityConstraintViolationException {
```